src/cases/components/case-task/case-task.component.ts

```diff
@@ -1,18 +1,39 @@
 import { Component, Input, OnInit } from '@angular/core';
-import { TaskPermission } from '../../../work-allocation-2/models/tasks/task-permission.model';
 import { UserInfo } from '../../../app/models';
 import { SessionStorageService } from '../../../app/services';
 import { Task } from '../../../work-allocation-2/models/tasks';
+import { TaskPermission } from '../../../work-allocation-2/models/tasks/task-permission.model';
 
 @Component({
   selector: 'exui-case-task',
   templateUrl: './case-task.component.html',
   styleUrls: ['./case-task.component.scss']
 })
 export class CaseTaskComponent implements OnInit {
-  @Input() public task: Task;
-  public manageOptions: Array<{text: string, path: string}>;
-  constructor(private readonly sessionStorageService: SessionStorageService) {}
+
+  constructor(private readonly sessionStorageService: SessionStorageService) {
+  }
+
+  public get task(): Task {
+    return this.pTask;
+  }
+
+  @Input()
+  public set task(value: Task) {
+    value.description = CaseTaskComponent.replaceCaseRefVarWithCaseId(value.description, value);
+    this.pTask = value;
+  }
+
+  private static CASE_REFERENCE_VARIABLE = '${[CASE_REFERENCE]}';
+  public manageOptions: { text: string, path: string }[];
+  private pTask: Task;
+
+  public static replaceCaseRefVarWithCaseId(str: string, task: Task): string {
```

Comment:
Unit tests for this function please?

Reply:
I have a unit test for the function check the `case-task.component.spec.ts`